tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,634 @@
+    of the ``godot-cpp`` folder.
+
+    You don't have to do it this way, but we've found it easiest to manage. If you
+    decide to download the repositories or clone them into your folder,
+    make sure to keep the folder layout the same as we've setup here. Much of
+    the code we'll be showcasing here assumes the project has this layout.
+
+If you cloned the example from the link specified in the introduction, the
+submodules are not automatically initialized. You will need to execute the
+following commands:
+
+.. code-block:: none
+
+    cd gdextension_cpp_example
+    git submodule update --init --recursive
+
+This will clone these two repositories into your project folder.
+
+Building the C++ bindings
+-------------------------
+
+Now that we've downloaded our prerequisites, it is time to build the C++
+bindings.
+
+The repository contains a copy of the metadata for the current Godot release,
+but if you need to build these bindings for a newer version of Godot, simply
+call the Godot executable:
+
+.. code-block:: none
+
+    godot --dump-extension-api extension_api.json
+
+Place the resulting ``extension_api.json`` file in the project folder and add
+``custom_api_file=<PATH_TO_FILE>`` to the scons command
+below.
+
+To generate and compile the bindings, use this command (replacing ``<platform>``
+with ``windows``, ``linux`` or ``macos`` depending on your OS):
+
+To speed up compilation, add `-jN` at the end of the SCons command line where `N`
+is the number of CPU threads you have on your system. The example below uses 4 threads.
+
+.. code-block:: none
+
+    cd godot-cpp
+    scons platform=<platform> -j4
+    cd ..
+
+This step will take a while. When it is completed, you should have static
+libraries that can be compiled into your project stored in ``godot-cpp/bin/``.
+
+.. note::
+
+    You may need to add ``bits=64`` to the command on Windows or Linux.
+
+Creating a simple plugin
+------------------------
+
+Now it's time to build an actual plugin. We'll start by creating an empty Godot
+project in which we'll place a few files.
+
+Open Godot and create a new project. For this example, we will place it in a
+folder called ``demo`` inside our GDExtension module's folder structure.
+
+In our demo project, we'll create a scene containing a Node called "Main" and
+we'll save it as ``main.tscn``. We'll come back to that later.
+
+Back in the top-level GDExtension module folder, we're also going to create a
+subfolder called ``src`` in which we'll place our source files.
+
+You should now have ``demo``, ``godot-cpp``, and ``src``
+directories in your GDExtension module.
+
+In the ``src`` folder, we'll start with creating our header file for the
+GDExtension node we'll be creating. We will name it ``gdexample.h``:
+
+.. code-block:: C++
+
+    #ifndef GDEXAMPLE_H
+    #define GDEXAMPLE_H
+
+    #include <sprite2d.hpp>
+
+    namespace godot {
+
+    class GDExample : public Sprite2D {
+        GD_CLASS(GDExample, Sprite2D)
```

Comment:
```suggestion
        GDCLASS(GDExample, Sprite2D)
```

I think you meant `GDCLASS`. I got stuck here for a little while 🤓

Reply:
@yudi-azvd Thanks for noticing! It's updated now.